### ISSUE TITLE
Failed to find network for networkMode managed / question about hybrid-cloud

### ISSUE THREAD

**Issue by perlun** [rancher/rancher#11420]
**Rancher versions:**
rancher/server: 1.6.14
rancher/agent: 1.2.9

**Infrastructure Stack versions:**
healthcheck: rancher/healthcheck:v0.3.3
ipsec: rancher/net:v0.13.7
network-services: rancher/network-manager:v0.7.19
scheduler: rancher/scheduler:v0.8.3
kubernetes (if applicable): N/A

**Docker version: (`docker version`,`docker info` preferred)**
```
Client:
 Version:	17.12.0-ce
 API version:	1.35
 Go version:	go1.9.2
 Git commit:	c97c6d6
 Built:	Wed Dec 27 20:11:19 2017
 OS/Arch:	linux/amd64

Server:
 Engine:
  Version:	17.12.0-ce
  API version:	1.35 (minimum version 1.12)
  Go version:	go1.9.2
  Git commit:	c97c6d6
  Built:	Wed Dec 27 20:09:53 2017
  OS/Arch:	linux/amd64
  Experimental:	false
```

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
```
NAME="Ubuntu"
VERSION="16.04.3 LTS (Xenial Xerus)"
```

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
Hybrid-cloud, server in AWS and nodes on-premise

**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
External DB

**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle

**Steps to Reproduce:**
Hi,

Thanks for an interesting and very promising product. I'm currently in the
process of setting up Rancher as an internal proof-of-concept, but running into
some issues:

- When running the server and agent on a local Ubuntu VM, all works fine - I can add services and they start up correctly, view logs, run a shell in them etc.
- When running the server on Amazon EC2 and the agent in my local VM, it doesn't work at all - the agent can reach the server, but it fails to start up its Cattle services as needed.
- When running the agent on another EC2 server, it works well.
- When running the agent on a customer server, it also works well.

Here are the logs for the first scenario, when it works (both server and client in Ubuntu VM, "on-premise"):

<details>
<summary>Click to expand <code>rancher-agent</code> logs</summary>
<pre>
time="2018-02-14T06:40:54Z" level=info msg="Launching agent"
time="2018-02-14T06:40:54Z" level=info msg="Detecting cloud provider"
time="2018-02-14T06:40:54Z" level=info msg="Listening for events on http://10.211.55.7:8080/v1"
time="2018-02-14T06:40:54Z" level=info msg="launching hostapi"
time="2018-02-14T06:40:54Z" level=warning msg="stat /var/lib/rancher/state/info.json: no such file or directory"
time="2018-02-14T06:40:54Z" level=warning msg="stat /var/lib/rancher/state/info.json: no such file or directory"
time="2018-02-14T06:40:54Z" level=info msg="Starting event router."
time="2018-02-14T06:40:54Z" level=info msg="Serialize compute requests: false, driver: overlay2"
time="2018-02-14T06:40:54Z" level=info msg="Initializing event router" workerCount=250
time="2018-02-14T06:40:54Z" level=info msg="Connecting to proxy." url="ws://10.211.55.7:8080/v1/connectbackend?token=<removed>"
time="2018-02-14T06:40:54Z" level=info msg="Connection established"
time="2018-02-14T06:40:54Z" level=info msg="Starting websocket pings"
time="2018-02-14T06:40:55Z" level=info msg="Received event: Name: storage.image.activate, Event Id: d914d0cd-36b0-40a1-9844-af9de2776a1d, Resource Id: 1ispm124"
time="2018-02-14T06:40:55Z" level=info msg="Reply: d914d0cd-36b0-40a1-9844-af9de2776a1d, storage.image.activate, 1ispm124:imageStoragePoolMap"
time="2018-02-14T06:40:55Z" level=info msg="Received event: Name: storage.volume.activate, Event Id: 484db3b9-1172-40f9-8440-effae7de7098, Resource Id: 1vspm162"
time="2018-02-14T06:40:55Z" level=info msg="Reply: 484db3b9-1172-40f9-8440-effae7de7098, storage.volume.activate, 1vspm162:volumeStoragePoolMap"
time="2018-02-14T06:40:55Z" level=info msg="Received event: Name: compute.instance.activate, Event Id: 21ac8499-45ff-4de1-952f-27233ec9d23a, Resource Id: 1ihm130"
time="2018-02-14T06:40:56Z" level=info msg="rancher id [132]: Container with docker id [72cb0b1f7246465eb95f306cf88bbfcc9dfd77b3ad5c50198761a59d446f878f] has been started"
time="2018-02-14T06:40:56Z" level=info msg="Reply: 21ac8499-45ff-4de1-952f-27233ec9d23a, compute.instance.activate, 1ihm130:instanceHostMap"
time="2018-02-14T06:41:09Z" level=error msg="retry check cloud provider" count=1 error="EC2MetadataRequestError: failed to get EC2 instance identity document\ncaused by: RequestError: send request failed\ncaused by: Get http://169.254.169.254/latest/dynamic/instance-identity/document: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)"
time="2018-02-14T06:41:33Z" level=error msg="retry check cloud provider" count=1 error="Get http://100.100.100.200/latest/meta-data/zone-id: dial tcp 100.100.100.200:80: getsockopt: connection refused"
time="2018-02-14T06:42:05Z" level=info msg="Received event: Name: storage.volume.activate, Event Id: e3b7cf46-584a-4ddb-a890-4ca0f2a5e32d, Resource Id: 1vspm162"
time="2018-02-14T06:42:05Z" level=info msg="Reply: e3b7cf46-584a-4ddb-a890-4ca0f2a5e32d, storage.volume.activate, 1vspm162:volumeStoragePoolMap"
time="2018-02-14T06:42:05Z" level=info msg="Received event: Name: compute.instance.activate, Event Id: 821f1a6f-1509-4f07-8860-fdbfe96c4d16, Resource Id: 1ihm130"
time="2018-02-14T06:42:05Z" level=info msg="rancher id [132]: Container with docker id [72cb0b1f7246465eb95f306cf88bbfcc9dfd77b3ad5c50198761a59d446f878f] has been started"
time="2018-02-14T06:42:05Z" level=info msg="Reply: 821f1a6f-1509-4f07-8860-fdbfe96c4d16, compute.instance.activate, 1ihm130:instanceHostMap"
time="2018-02-14T06:42:07Z" level=info msg="Received event: Name: compute.instance.deactivate, Event Id: abe82383-610f-4363-8d6b-f890db37096a, Resource Id: 1ihm129"
time="2018-02-14T06:42:07Z" level=info msg="Reply: abe82383-610f-4363-8d6b-f890db37096a, compute.instance.deactivate, 1ihm129:instanceHostMap"
time="2018-02-14T06:42:07Z" level=info msg="Received event: Name: storage.volume.remove, Event Id: 73988a7b-6f67-4b95-b04e-b109f776a45b, Resource Id: 1vspm161"
time="2018-02-14T06:42:07Z" level=info msg="rancher id [1vspm161]: Volume with name [] has been removed"
time="2018-02-14T06:42:07Z" level=info msg="Reply: 73988a7b-6f67-4b95-b04e-b109f776a45b, storage.volume.remove, 1vspm161:volumeStoragePoolMap"
time="2018-02-14T06:42:08Z" level=warning msg="stat /var/lib/rancher/state/info.json: no such file or directory"
time="2018-02-14T06:42:08Z" level=warning msg="Hit websocket pong timeout. Last websocket ping received at 2018-02-14 06:41:34.824539502 +0000 UTC. Closing connection."
time="2018-02-14T06:42:08Z" level=error msg="Received error reading from socket. Exiting." error="read tcp 10.211.55.7:55066->10.211.55.7:8080: use of closed network connection"
time="2018-02-14T06:42:08Z" level=error msg="Failed to connect to websocket proxy: %vread tcp 10.211.55.7:55066->10.211.55.7:8080: use of closed network connection"
time="2018-02-14T06:42:13Z" level=info msg="Connecting to proxy." url="ws://10.211.55.7:8080/v1/connectbackend?token=eyJhbGciOiJSUzI1NiJ9.eyJyZXBvcnRlZFV1aWQiOiJjMjZhZWMxYy1kMjhkLTQ1YjMtNDc0ZS0zODRlZmY2ZDgyMTIiLCJzdWIiOiJjYXR0bGUiLCJraWQiOiJkZWZhdWx0IiwiaXNzIjoiaHR0cDpcL1wvY2F0dGxlLmlvIiwiZXhwIjoxNTE4NTkwODMzLCJpYXQiOjE1MTg1OTA1MzN9.GBXVlPL823gTxFHJV4O4czWunlw-n2Nja4ZF5qOCk-nqGl-HDLfvCNA7U9S2yohp3bEEU5z7hXzA1O48bxxWZvePB9flk0CoFoKqQSZYgAAtj4U0Xcj6AIMtfjH3U1xmRh-M0sIM6YW46cKIkFoNolUYRFtCFP3zWkJRaPX4mweHeTUVmZybf7gMItfB-qHV59ySqL06wMYQ4ZQZREIF1PvaLVssduJ-4zHEX5H_GZ2BPJshB043ktN7VW5Yga7YiKTPrT018_Lrz3LZaT5Vmvy5coNWe0D4UPp1jj94wlGLSbs8--n5ojQV6Fo_orgXzHSfbxZ7YAt6y1Y_M770Pg"
time="2018-02-14T06:42:20Z" level=error msg="retry check cloud provider" count=2 error="EC2MetadataRequestError: failed to get EC2 instance identity document\ncaused by: RequestError: send request failed\ncaused by: Get http://169.254.169.254/latest/dynamic/instance-identity/document: dial tcp 169.254.169.254:80: getsockopt: no route to host"
time="2018-02-14T06:42:31Z" level=warning msg="stat /var/lib/rancher/state/info.json: no such file or directory"
time="2018-02-14T06:42:50Z" level=warning msg="stat /var/lib/rancher/state/info.json: no such file or directory"
time="2018-02-14T06:43:02Z" level=error msg="retry check cloud provider" count=3 error="EC2MetadataRequestError: failed to get EC2 instance identity document\ncaused by: RequestError: send request failed\ncaused by: Get http://169.254.169.254/latest/dynamic/instance-identity/document: dial tcp 169.254.169.254:80: getsockopt: no route to host"
time="2018-02-14T06:43:32Z" level=warning msg="stat /var/lib/rancher/state/info.json: no such file or directory"
time="2018-02-14T06:43:44Z" level=error msg="retry check cloud provider" count=4 error="EC2MetadataRequestError: failed to get EC2 instance identity document\ncaused by: RequestError: send request failed\ncaused by: Get http://169.254.169.254/latest/dynamic/instance-identity/document: dial tcp 169.254.169.254:80: getsockopt: no route to host"
time="2018-02-14T06:44:14Z" level=warning msg="stat /var/lib/rancher/state/info.json: no such file or directory"
time="2018-02-14T06:44:27Z" level=error msg="retry check cloud provider" count=5 error="EC2MetadataRequestError: failed to get EC2 instance identity document\ncaused by: RequestError: send request failed\ncaused by: Get http://169.254.169.254/latest/dynamic/instance-identity/document: dial tcp 169.254.169.254:80: getsockopt: no route to host"
time="2018-02-14T06:44:57Z" level=warning msg="stat /var/lib/rancher/state/info.json: no such file or directory"
time="2018-02-14T06:45:01Z" level=error msg="checking aliyun cloud provider error after 2 attempts, last error: Get http://100.100.100.200/latest/meta-data/zone-id: dial tcp 100.100.100.200:80: i/o timeout"
time="2018-02-14T06:45:09Z" level=error msg="checking aws cloud provider error after 6 attempts, last error: EC2MetadataRequestError: failed to get EC2 instance identity document\ncaused by: RequestError: send request failed\ncaused by: Get http://169.254.169.254/latest/dynamic/instance-identity/document: dial tcp 169.254.169.254:80: getsockopt: no route to host"
time="2018-02-14T07:11:44Z" level=error msg="Error writing EOT message." error="write unix @->/var/run/docker.sock: write: broken pipe"
time="2018-02-14T07:12:06Z" level=error msg="Error with the container log scanner." error="context canceled"
</pre>
</details>
<br/>

I then launch the on-premise agent like this:

```
$ sudo docker run --rm --privileged -v /var/run/docker.sock:/var/run/docker.sock -v /var/lib/rancher-aws:/var/lib/rancher rancher/agent:v1.2.9 http://my-fqdn.net:8080/v1/scripts/473804943071BE613A88:1514678400000:6wsdmwQj3hOWUwLTQ9O4tVpUur4

INFO: Running Agent Registration Process, CATTLE_URL=http://my-fqdn.net:8080/v1
INFO: Attempting to connect to: http://my-fqdn.net:8080/v1
INFO: http://my-fqdn.net:8080/v1 is accessible
INFO: Inspecting host capabilities
INFO: Boot2Docker: false
INFO: Host writable: true
INFO: Token: xxxxxxxx
INFO: Running registration
INFO: Printing Environment
INFO: ENV: CATTLE_ACCESS_KEY=67FF3E8D85EFA83A6E72
INFO: ENV: CATTLE_HOME=/var/lib/cattle
INFO: ENV: CATTLE_REGISTRATION_ACCESS_KEY=registrationToken
INFO: ENV: CATTLE_REGISTRATION_SECRET_KEY=xxxxxxx
INFO: ENV: CATTLE_SECRET_KEY=xxxxxxx
INFO: ENV: CATTLE_URL=http://my-fqdn.net:8080/v1
INFO: ENV: DETECTED_CATTLE_AGENT_IP=81.209.9.65
INFO: ENV: RANCHER_AGENT_IMAGE=rancher/agent:v1.2.9
INFO: Deleting container rancher-agent
INFO: Launched Rancher Agent: e822ddb6cab4fb413c5fe263982a731a6aaed1d22c4ba13fac9aff0c8512a43a
```

However, when this agent has been started only a very few containers are running on it. All the Cattle infrastructure services are missing:

![image](https://user-images.githubusercontent.com/630613/36195426-1a81fc6e-1176-11e8-9753-ae54e9405dfb.png)

When I try to add a service to the Default stack, I get this error:

`Failed to allocate instance [container:1i113]: Bad instance [container:1i113] in state [removed]: Failed to find network for networkMode managed)`

Here are the logs:

<details>
<summary>Click to expand <code>rancher-agent</code> logs</summary>
<pre>
2/14/2018 9:32:01 AMtime="2018-02-14T07:32:01Z" level=info msg="Connecting to proxy." url="ws://my-fqdn.net:8080/v1/connectbackend?token=<removed>"
2/14/2018 9:32:01 AMtime="2018-02-14T07:32:01Z" level=info msg="Received event: Name: config.update, Event Id: f7ad7a5e-8b42-43d0-af3a-57e81db0e148, Resource Id: 1a11"
2/14/2018 9:32:10 AMtime="2018-02-14T07:32:10Z" level=info msg="Received event: Name: compute.instance.inspect, Event Id: 2248a79e-ff05-4139-a7ad-f7d0d7c5e49a, Resource Id: "
2/14/2018 9:32:10 AMtime="2018-02-14T07:32:10Z" level=info msg="Received event: Name: compute.instance.inspect, Event Id: aad85c71-c291-45b2-b5b3-2f6dcbaa40b3, Resource Id: "
2/14/2018 9:32:10 AMtime="2018-02-14T07:32:10Z" level=info msg="rancher id []: Container with docker id [2ec88bbcf3056d4a34b0f976d629f913cfd5507712b57a2dcdc8c117fb11f1de] has been inspected"
2/14/2018 9:32:10 AMtime="2018-02-14T07:32:10Z" level=info msg="rancher id []: Container with docker id [cd6d6763578d16b994055c1ee67b11499eff4834a5dc0a0a382b2c0db2f5a609] has been inspected"
2/14/2018 9:32:10 AMtime="2018-02-14T07:32:10Z" level=info msg="Reply: 2248a79e-ff05-4139-a7ad-f7d0d7c5e49a, compute.instance.inspect, :instanceInspect"
2/14/2018 9:32:10 AMtime="2018-02-14T07:32:10Z" level=info msg="Reply: aad85c71-c291-45b2-b5b3-2f6dcbaa40b3, compute.instance.inspect, :instanceInspect"
2/14/2018 9:32:10 AMtime="2018-02-14T07:32:10Z" level=info msg="Received event: Name: storage.volume.activate, Event Id: dae9252a-104c-46eb-a09c-7108efe2034e, Resource Id: 1vspm77"
2/14/2018 9:32:10 AMtime="2018-02-14T07:32:10Z" level=info msg="Received event: Name: storage.volume.activate, Event Id: 3c8da638-7c9f-4a8d-a2bd-b1eedf633d82, Resource Id: 1vspm78"
2/14/2018 9:32:10 AMtime="2018-02-14T07:32:10Z" level=info msg="Reply: dae9252a-104c-46eb-a09c-7108efe2034e, storage.volume.activate, 1vspm77:volumeStoragePoolMap"
2/14/2018 9:32:10 AMtime="2018-02-14T07:32:10Z" level=info msg="Reply: 3c8da638-7c9f-4a8d-a2bd-b1eedf633d82, storage.volume.activate, 1vspm78:volumeStoragePoolMap"
2/14/2018 9:32:10 AMtime="2018-02-14T07:32:10Z" level=info msg="Received event: Name: compute.instance.activate, Event Id: 69e039b3-b4af-4b86-8170-791bc0e1271a, Resource Id: 1ihm28"
2/14/2018 9:32:10 AMtime="2018-02-14T07:32:10Z" level=info msg="Reply: 69e039b3-b4af-4b86-8170-791bc0e1271a, compute.instance.activate, 1ihm28:instanceHostMap"
2/14/2018 9:32:10 AMtime="2018-02-14T07:32:10Z" level=info msg="Received event: Name: compute.instance.activate, Event Id: 7a6b3557-a4b9-4984-a1b8-d6d8e6c23727, Resource Id: 1ihm29"
2/14/2018 9:32:10 AMtime="2018-02-14T07:32:10Z" level=info msg="Reply: 7a6b3557-a4b9-4984-a1b8-d6d8e6c23727, compute.instance.activate, 1ihm29:instanceHostMap"
2/14/2018 9:32:10 AMtime="2018-02-14T07:32:10Z" level=error msg="retry check cloud provider" count=1 error="EC2MetadataRequestError: failed to get EC2 instance identity document\ncaused by: RequestError: send request failed\ncaused by: Get http://169.254.169.254/latest/dynamic/instance-identity/document: dial tcp 169.254.169.254:80: getsockopt: no route to host"
2/14/2018 9:32:11 AMtime="2018-02-14T07:32:11Z" level=info msg="Received event: Name: storage.volume.activate, Event Id: 98df08e0-c0b5-4905-8db2-07167446bca0, Resource Id: 1vspm83"
2/14/2018 9:32:11 AMtime="2018-02-14T07:32:11Z" level=info msg="Reply: 98df08e0-c0b5-4905-8db2-07167446bca0, storage.volume.activate, 1vspm83:volumeStoragePoolMap"
2/14/2018 9:32:11 AMtime="2018-02-14T07:32:11Z" level=info msg="Received event: Name: storage.volume.activate, Event Id: 6cfc2776-f261-49d5-bbdc-3fe3d55c2983, Resource Id: 1vspm81"
2/14/2018 9:32:11 AMtime="2018-02-14T07:32:11Z" level=info msg="Received event: Name: storage.volume.activate, Event Id: 4a6faf0b-072a-46f7-b3d9-51c385271d8b, Resource Id: 1vspm85"
2/14/2018 9:32:11 AMtime="2018-02-14T07:32:11Z" level=info msg="Reply: 4a6faf0b-072a-46f7-b3d9-51c385271d8b, storage.volume.activate, 1vspm85:volumeStoragePoolMap"
2/14/2018 9:32:11 AMtime="2018-02-14T07:32:11Z" level=info msg="Reply: 6cfc2776-f261-49d5-bbdc-3fe3d55c2983, storage.volume.activate, 1vspm81:volumeStoragePoolMap"
2/14/2018 9:32:11 AMtime="2018-02-14T07:32:11Z" level=info msg="Received event: Name: storage.volume.activate, Event Id: 3bf9dcab-e6da-48b5-8b03-d0d431e65236, Resource Id: 1vspm82"
2/14/2018 9:32:11 AMtime="2018-02-14T07:32:11Z" level=info msg="Reply: 3bf9dcab-e6da-48b5-8b03-d0d431e65236, storage.volume.activate, 1vspm82:volumeStoragePoolMap"
2/14/2018 9:32:11 AMtime="2018-02-14T07:32:11Z" level=info msg="Received event: Name: storage.volume.activate, Event Id: 83254cbb-630c-4416-9488-994829806f8b, Resource Id: 1vspm84"
2/14/2018 9:32:11 AMtime="2018-02-14T07:32:11Z" level=info msg="Reply: 83254cbb-630c-4416-9488-994829806f8b, storage.volume.activate, 1vspm84:volumeStoragePoolMap"
2/14/2018 9:32:11 AMtime="2018-02-14T07:32:11Z" level=info msg="Received event: Name: storage.volume.activate, Event Id: e36eb0a4-9896-46f6-af08-054c26081166, Resource Id: 1vspm79"
2/14/2018 9:32:11 AMtime="2018-02-14T07:32:11Z" level=info msg="Reply: e36eb0a4-9896-46f6-af08-054c26081166, storage.volume.activate, 1vspm79:volumeStoragePoolMap"
2/14/2018 9:32:11 AMtime="2018-02-14T07:32:11Z" level=info msg="Received event: Name: storage.volume.activate, Event Id: c223feae-fdd2-4492-9d88-15d44f1cb1c9, Resource Id: 1vspm80"
2/14/2018 9:32:11 AMtime="2018-02-14T07:32:11Z" level=info msg="Reply: c223feae-fdd2-4492-9d88-15d44f1cb1c9, storage.volume.activate, 1vspm80:volumeStoragePoolMap"
2/14/2018 9:32:11 AMtime="2018-02-14T07:32:11Z" level=info msg="Received event: Name: storage.volume.activate, Event Id: e40b4696-35c2-46d4-b5f8-e945e5b599f7, Resource Id: 1vspm86"
2/14/2018 9:32:11 AMtime="2018-02-14T07:32:11Z" level=info msg="Reply: e40b4696-35c2-46d4-b5f8-e945e5b599f7, storage.volume.activate, 1vspm86:volumeStoragePoolMap"
2/14/2018 9:32:11 AMtime="2018-02-14T07:32:11Z" level=info msg="Received event: Name: storage.volume.activate, Event Id: 7224b49a-867d-4b9b-8761-ac8d05eda299, Resource Id: 1vspm87"
2/14/2018 9:32:11 AMtime="2018-02-14T07:32:11Z" level=info msg="Reply: 7224b49a-867d-4b9b-8761-ac8d05eda299, storage.volume.activate, 1vspm87:volumeStoragePoolMap"
2/14/2018 9:32:11 AMtime="2018-02-14T07:32:11Z" level=info msg="Received event: Name: storage.volume.activate, Event Id: 3322130a-b070-48c3-80c8-b0e3860efad3, Resource Id: 1vspm88"
2/14/2018 9:32:11 AMtime="2018-02-14T07:32:11Z" level=info msg="Reply: 3322130a-b070-48c3-80c8-b0e3860efad3, storage.volume.activate, 1vspm88:volumeStoragePoolMap"
2/14/2018 9:32:36 AMtime="2018-02-14T07:32:36Z" level=error msg="Error with the container log scanner." error="context canceled"
2/14/2018 9:32:40 AMtime="2018-02-14T07:32:40Z" level=warning msg="stat /var/lib/rancher/state/info.json: no such file or directory"
2/14/2018 9:32:53 AMtime="2018-02-14T07:32:53Z" level=error msg="retry check cloud provider" count=2 error="EC2MetadataRequestError: failed to get EC2 instance identity document\ncaused by: RequestError: send request failed\ncaused by: Get http://169.254.169.254/latest/dynamic/instance-identity/document: dial tcp 169.254.169.254:80: getsockopt: no route to host"
2/14/2018 9:33:23 AMtime="2018-02-14T07:33:23Z" level=warning msg="stat /var/lib/rancher/state/info.json: no such file or directory"
2/14/2018 9:33:35 AMtime="2018-02-14T07:33:35Z" level=error msg="retry check cloud provider" count=3 error="EC2MetadataRequestError: failed to get EC2 instance identity document\ncaused by: RequestError: send request failed\ncaused by: Get http://169.254.169.254/latest/dynamic/instance-identity/document: dial tcp 169.254.169.254:80: getsockopt: no route to host"
2/14/2018 9:34:05 AMtime="2018-02-14T07:34:05Z" level=warning msg="stat /var/lib/rancher/state/info.json: no such file or directory"
2/14/2018 9:34:17 AMtime="2018-02-14T07:34:17Z" level=error msg="retry check cloud provider" count=4 error="EC2MetadataRequestError: failed to get EC2 instance identity document\ncaused by: RequestError: send request failed\ncaused by: Get http://169.254.169.254/latest/dynamic/instance-identity/document: dial tcp 169.254.169.254:80: getsockopt: no route to host"
2/14/2018 9:34:28 AMtime="2018-02-14T07:34:28Z" level=error msg="retry check cloud provider" count=1 error="Get http://100.100.100.200/latest/meta-data/zone-id: dial tcp 100.100.100.200:80: i/o timeout"
2/14/2018 9:34:47 AMtime="2018-02-14T07:34:47Z" level=warning msg="stat /var/lib/rancher/state/info.json: no such file or directory"
2/14/2018 9:34:58 AMtime="2018-02-14T07:34:58Z" level=warning msg="stat /var/lib/rancher/state/info.json: no such file or directory"
2/14/2018 9:34:59 AMtime="2018-02-14T07:34:59Z" level=error msg="retry check cloud provider" count=5 error="EC2MetadataRequestError: failed to get EC2 instance identity document\ncaused by: RequestError: send request failed\ncaused by: Get http://169.254.169.254/latest/dynamic/instance-identity/document: dial tcp 169.254.169.254:80: getsockopt: no route to host"
2/14/2018 9:35:30 AMtime="2018-02-14T07:35:30Z" level=warning msg="stat /var/lib/rancher/state/info.json: no such file or directory"
2/14/2018 9:35:42 AMtime="2018-02-14T07:35:42Z" level=error msg="checking aws cloud provider error after 6 attempts, last error: EC2MetadataRequestError: failed to get EC2 instance identity document\ncaused by: RequestError: send request failed\ncaused by: Get http://169.254.169.254/latest/dynamic/instance-identity/document: dial tcp 169.254.169.254:80: getsockopt: no route to host"
</pre>
</details>
<br/>

I tried launching the agent on another AWS EC2 server, which worked fine (it
accessed the http://my-fqdn.net:8080 address using its public IP address.) - it
started all the infrastructure services without any problems.

I launched the agent on a customer server, which also worked well - the
infrastructure services started without problems.

----

However, while investigating this I now read on [this web page](http://rancher.com/docs/rancher/latest/en/hosts/custom/) the following:

> All the hosts within Rancher will need to be on the same network as Rancher server.

**Is this really true???** I was under the impression that the hosts only need to
be able to access the Rancher server (using http and websockets), i.e. a
hybrid-cloud setup with a central Rancher server and nodes within "private"
on-premise networks should be possible.

This does indeed _work_ so I wonder... is the information on this web page
outdated or am I missing something?

**Comment by perlun** [comment]
The [quick start guide](http://rancher.com/docs/rancher/v1.6/en/quick-start-guide/) gives a different picture:

> To add a host, access the UI and click Infrastructure, which will immediately bring you to the Hosts page. Click on the Add Host. Rancher will prompt you to select a host registration URL. This URL is where Rancher server is running and must be reachable from all the hosts that you will be adding. This is useful in installations where Rancher server will be exposed to the Internet through a NAT firewall or a load balancer. If your host has a private or local IP address like 192.168.*.*, Rancher will print a warning asking you to make sure that the hosts can indeed reach the URL.

Which way is it; do the Rancher hosts need be on the same _network_ as the server or merely be able to access it using HTTP(S) (including websockets)?

**Comment by vincent99** [comment]
That note is referring specifically to using (only) private IPs on the VMs deployed with docker-machine.  For docker-machine provisioning the server container SSHes to the IP assigned to the machine to install docker.  So if your server is on a random different network, it can't reach the VM to finish setting it up. 

The hosts need to be able to connect to the server container (with websockets) for management.  And all hosts in an environment need unique, mutually-reachable IP addresses and 500+4500/udp open between all combinations of each other for the managed IPSec network (server container is not involved).

**Comment by perlun** [comment]
> That note is referring specifically to using (only) private IPs on the VMs deployed with docker-machine. For docker-machine provisioning the server container SSHes to the IP assigned to the machine to install docker. So if your server is on a random different network, it can't reach the VM to finish setting it up.

Ah, that makes complete sense... Silly me! 🤦‍♂️ 

> The hosts need to be able to connect to the server container (with websockets) for management. And all hosts in an environment need unique, mutually-reachable IP addresses and 500+4500/udp open between all combinations of each other for the managed IPSec network (server container is not involved).

Thanks for clearing that up. Is managed networks the current recommendation, or host networking? I feel a bit afraid of the managed network approach since I've seen so many problems with overlay networking before (not w/ Rancher, with plain Docker controlled via Chef recipes.)

**Comment by superseb** [comment]
The ipsec or vxlan overlay is included by default, that's how the cross-host communication takes place and is created automatically between all hosts added to the environment. 

Regarding Infrastructure Services, they should be included by default when creating an environment from the default environment templates. If they are not coming up when adding a node, I would like to know the steps to reproduce to see if something can be improved there.

**Comment by perlun** [comment]
Thanks for your feedback @superseb. I tried again:

The environment in question is using the Cattle orchestration and template:

<img width="1097" alt="image" src="https://user-images.githubusercontent.com/630613/36297895-d3a7d104-12fd-11e8-9a43-56a5a89a3281.png">

<img width="1360" alt="image" src="https://user-images.githubusercontent.com/630613/36297903-dd714792-12fd-11e8-897f-9875596dca1e.png">

Another host in this environment is running fine, it got registered with all parts as required.

I now tred to add my Ubuntu 16.04 VM to this environment, using a command like this:

> $ sudo docker run --rm --privileged -v /var/run/docker.sock:/var/run/docker.sock -v /var/lib/rancher:/var/lib/rancher rancher/agent:v1.2.9 http://my-fqdn.net:8080/v1/scripts/<removed>:<removed>

Funnily enough, _this time it worked_. The parallels VM got registered without errors, all the infrastructure services started, and the node seemed to be fully working.

It must have been a glitch when I run it the first time; I removed all Rancher containers thoroughly now, as well as the `/var/lib/rancher` folders on the host (I had done it before also, but maybe some state slipped through anyway from my different tests.)

Sorry to have wasted your time, but thanks for the help with resolving this! Closing as a user issue.
